unity-client/Assets/Scripts/MainScripts/DCL/Controllers/BuilderMode/BuilderInWorldBridge/BuilderInWorldProtocol.cs

```diff
@@ -0,0 +1,132 @@
+using DCL.Models;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class BuilderInWorldProtocol 
+{
+    #region Class Declarations
+
+    [System.Serializable]
+    public class QuaternionRepresentantion
+    {
+        public float x;
+        public float y;
+        public float z;
+        public float w;
+
+        public QuaternionRepresentantion(Quaternion quaternion)
+        {
+
+            x = quaternion.x;
+            y = quaternion.y;
+            z = quaternion.z;
+            w = quaternion.w;
+        }
+    }
+
+    #region Components
+
+    [System.Serializable]
+    public abstract class GenericComponent
+    {
+
+    }
+
+    [System.Serializable]
+    public class TransformComponent : GenericComponent
+    {
+        public Vector3 position;
+
+        public QuaternionRepresentantion rotation;
```

Comment:
Why we need a representation here? `Quaternions` are serializable structs just as `Vector3`, this shouldn't be needed.

Reply:
The problem is that we need to use NewtonJson to serialize. NewtonJson finds circular dependencies, and if I ignore them with a JsonFormat, it will serialize all the info in Quaternions, this is done in order to avoid unnecessary info on the serialization